docs/workflow/argocd.md

```diff
@@ -0,0 +1,193 @@
+---
+title: "ArgoCD"
+toc: true
+weight: 591
+indent: true
+---
+
+# Continuous Delivery with ArgoCD
+
+[ArgoCD] is a continuous delivery tool for Kubernetes. It runs in a Kubernetes
+cluster and enables a GitOps workflow by watching repositories and automatically
+updating deployments using Kubernetes manifests. In this guide we will walk
+through setting up ArgoCD, installing Crossplane and a few cloud provider
+stacks, setting up an infrastructure pipeline, then finally deploying an
+application that consumes cloud provider resources.
+
+## Table of Contents
+
+1. [Pre-requisites](#pre-requisites)
+1. [Preparation](#preparation)
+1. [Set Up ArgoCD](#set-up-argocd)
+1. [Install Crossplane Components](#install-crossplane-components)
+1. [Set Up an Infrastructure Pipeline](#set-up-an-infrastructure-pipeline)
+    1. [Create an Infrastructure
+       Repository](#create-an-infrastructure-repository)
+    1. [Create an ArgoCD Project for
+       Infrastructure](#create-an-argocd-project-for-infrastructure)
+    1. [Configure Dev and Prod
+       Environments](#configure-dev-and-prod-environments)
+1. [Set Up an Application Pipeline](#set-up-an-application-pipeline)
+    1. [Create an Application Repository](#create-an-application-repository)
+    1. [Create an ArgoCD Project for
+       Applications](#create-an-argocd-project-for-applications)
+    1. [Configure an ArgoCD Application](#configure-an-argocd-application)
+    1. [Deploy](#deploy)
+1. [Clean Up](#clean-up)
+1. [Next Steps](#next-steps)
+
+## Pre-requisites
+
+This guide assumes the presence of:
+
+* An existing Kubernetes cluster
+* [kubectl] configured to access the cluster
+
+## Preparation
+
+Before we get started deploying infrastructure and application components, we
+need to set up ArgoCD and Crossplane in our existing Kubernetes cluster.
+
+### Set Up ArgoCD
+
+ArgoCD has a great [getting started] guide that walks through installing the
+platform into your Kubernetes cluster. We will only need to complete steps 1-3
+for this guide as we will be deploying all Crossplane components to the same
+cluster that is running ArgoCD.
+
+*Note: ArgoCD uses self-signed certificates, and it may be necessary to [enable
+insecure connections] on localhost to utilize port-forwarding.*
+
+### Install Crossplane Components
+
+For this guide we will assume that an organization wants to enable provisioning
+of managed service on [GCP], [AWS], and [Azure]. To install Crossplane along
+with the GCP, AWS, and Azure stacks, please see the [installation guide].
+
+## Set Up an Infrastructure Pipeline
+
+Now that ArgoCD and all Crossplane components are installed, it is time to build
+out our infrastructure environments. 
+
+### Create an Infrastructure Repository
```

Comment:
would add a getting started section to workflow.md basics, then link to that section from here (to orient new users):
### Environment Configuration
 * brief intro to repo layout
 * doing basic environment customization [patch.yaml](https://github.com/hasheddan/crosscd/blob/master/classpacks/services-guides/gcp/patch.yaml)
 * `kubectl apply -k` apply it

Reply:
@prasek thinking about making this a little more basic (i.e. more like the demo on TBS today). What do you think? How much of this do you still want moved to `workflow.md`?

Reply:
@hasheddan yes, the TBD demo was awesome! 

suggest going with something similar to https://github.com/crossplaneio/tbs/tree/master/episodes/5/assets/infra plus:
 - infra classes for each kind
 - env (dev, test) patch for kustomizing `region` and `network`

Then do this for each cloud:
```text
aws
└── app   # wordpress examples
└── env   # environments
       └── dev # ../../infra base + dev patch.yaml
       └── test # ../../infra base + test patch.yaml
└── infra  # classes for all kinds plus secure networking / conn

gcp
└── app
└── env 
       └── dev
       └── test
└── infra

azure
└── app
└── env 
       └── dev
       └── test
└── infra
```